### PR TITLE
Fix several NX-OS workflow issues

### PR DESCRIPTION
**Pull request by mikewiebe**
### What does this PR do?

* Updates platform support matrix
* More graceful handling of errors over nx-os device reloads and reboots.
* Fixes a bug when using `ssh` transport.  If a configuration file is passed to the nxos device that has the `description` field it will interpret subsequent commands as part of the `description` instead of additional nx-os commands.
* Fixes a payload ordering problem in the nxapi client code.

### Tests written?

Manually Tested using several state files.

Tested all 3 workflows.

* Proxy SSH Minion
* Proxy NX-API Minion
* GuestShell Minion

### Commits signed with GPG?

No

### CODE REVIEW

File: salt/modules/nxos.py [hunk 1]
```diff
@@ -338,11 +339,15 @@ def sendline(command, method='cli_show_ascii', **kwargs):
...
-        return _nxapi_request(command, method, **kwargs)
+    try:
+        if salt.utils.platform.is_proxy():
+            return __proxy__['nxos.sendline'](command, method, **kwargs)
+        else:
+            return _nxapi_request(command, method, **kwargs)
+    except socket_error as e:
+        return e.strerror
+    except NxosError as e:
+        return e.strerror
```
Comment (chrisvanheuveln) [inline salt/modules/nxos.py]: fyi: you can combine these exceptions:
```
    except (socket_error, NxosError) as e:
        return e.strerror
```

Comment (mikewiebe) [reply salt/modules/nxos.py]: Yeah I know.  I intentionally left them separate if we need to handle SSH vs NX-API errors differently for new use cases.  If you feel strongly I can combine them though for now.

Comment (chrisvanheuveln) [reply salt/modules/nxos.py]: It's fine. Don't change it.

**Comment by mikewiebe** [comment]
@dwoz Reaching out to you because I reached out to you in the past and it seemed to get some traction :).  This PR has been open since Apr18.  Fairly minor updates.  Is here anything else needed to merge this?  Thanks!

**Review by twangboy — APPROVED** [review]

**Comment by mikewiebe** [comment]
Closing this PR since these diffs will be merged into the `master` branch as part of https://github.com/saltstack/salt/pull/54931.

**Pull request closed without merging**